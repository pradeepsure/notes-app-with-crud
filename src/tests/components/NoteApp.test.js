import NotesApp from '../../components/NotesApp';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

let wrapper, notes, addNote, removeNote;
beforeEach(() => {
    notes = [{
        "noteTitle": "paytmDiscount",
        "noteDescription": "cashback upto 20 Rupees hurry",
        "id": 1
    }]
    addNote = jest.fn();
    removeNote = jest.fn();
    wrapper = shallow(
        <NotesApp
            notes={notes}
            addNote={addNote}
            removeNote={removeNote}
        />
    );
});


test('Render NotesApp component with notes correctly', () => {
    expect(wrapper).toMatchSnapshot();
});