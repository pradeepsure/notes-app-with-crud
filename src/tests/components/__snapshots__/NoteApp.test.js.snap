// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NotesApp component with notes correctly 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(NoteTaker) />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <NotesContainer
      notes={
        Array [
          Object {
            "id": 1,
            "noteDescription": "cashback upto 20 Rupees hurry",
            "noteTitle": "paytmDiscount",
          },
        ]
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
